sheet_to_triples/tests/test_run.py

```diff
@@ -0,0 +1,384 @@
+# Copyright 2021 Visual Meaning Ltd
+# This is free software licensed as GPL-3.0-or-later - see COPYING for terms.
+
+"""Unittests for the run.py module of sheet-to-triples."""
+
+import io
+import json
+import unittest
+
+import rdflib
+
+from unittest import mock
+
+from .. import run, trans
+
+
+def _mock_open(data):
+    return mock.patch('builtins.open', mock.mock_open(read_data=data))
+
+
+def _mock_basename():
+    return mock.patch(
+        'os.path.basename',
+        side_effect=lambda x: '/basename/' + x,
+    )
+
+
+def _mock_xl_load_book():
+    return mock.patch(
+        'sheet_to_triples.xl.load_book',
+        side_effect=lambda x: '/load_book/' + x,
+    )
+
+
+def _mock_xl_iter_sheet(test_iter):
+    return mock.patch(
+        'sheet_to_triples.xl.iter_sheet',
+        return_value=test_iter
+    )
+
+
+class StubArgs:
+    """Stub class with attribute values matching args keys."""
+    def __init__(self, args):
+        for arg in args:
+            setattr(self, arg, args[arg])
```

Comment:
Shouldn't need this, can use `argparse.Namespace` in pretty much the way you want, it just takes kwargs.